src/magic_folder/test/eliotutil.py

```diff
@@ -69,16 +69,14 @@ def run_and_republish(self, *a, **kw):
         from eliot._output import _DEFAULT_LOGGER as default_logger
 
         def republish():
-            # This is called as a cleanup function after capture_logging has
-            # restored the global/default logger to its original state.  We
-            # can now emit messages that go to whatever global destinations
-            # are installed.
-
             # The memory logger captures both the raw messages and the
             # serializer. We pass them both to the global logger, so that the
             # expected serialization is preserved.
             logger = storage.logger
             for msg, serializer in zip(logger.messages, logger.serializers):
+                # We copy the message here, as the cleanup function that
+                # capture_logging installs will also try to serialize all
+                # messages.
                 default_logger.write(msg, serializer)
 
             # And now that we've re-published all of the test's messages, we
```

Comment:
I think the `action.finish` needs to be moved til after the default logger is restored

Reply:
Or, we need to pass an explicit logger into the action start.